core/trino-main/src/main/java/io/trino/operator/output/PartitionedOutputOperator.java

```diff
@@ -28,6 +28,7 @@
 import io.trino.operator.OutputFactory;
```

Comment:
Could you group JMH results in `before/after` pairwise instead.

Add high level description of algorithm in commit description